lambdas/src/apis/collections/controllers/collections.ts

```diff
@@ -126,26 +133,30 @@ export async function contentsImage(client: SmartContentClient, req: Request, re
     let image = await client.downloadContent(hash)
     image = await resize(image).toBuffer()
 
-    const imageFilePath = getRarityImagePath(wearableMetadata)
+    const imageFilePath = getRarityImagePath(wearableMetadata.rarity)
     const finalImage = await resize(imageFilePath)
       .composite([{ input: image }])
       .toBuffer()
 
-    res.send(finalImage)
-
-    res.writeHead(200, {
-      'Content-Type': 'arraybuffer',
-      ETag: urn,
-      'Access-Control-Expose-Headers': '*',
-      'Cache-Control': 'public, max-age=31536000, immutable'
-    })
+    sendImageBuffer(res, finalImage, urn)
   } catch (e) {
     res.status(500).send(e.message)
   }
 }
 
-function getRarityImagePath(wearableMetadata: WearableMetadata) {
-  return `lambdas/resources/${wearableMetadata.rarity}.png`
+function sendImageBuffer(res: Response, image: Buffer, urn: string) {
+  res.send(image)
+
+  res.writeHead(200, {
+    'Content-Type': 'arraybuffer',
+    ETag: urn,
```

Comment:
`ETag` spec requires the string to be double quoted and it must include the variant size

```diff
-ETag: urn
+ETag: JSON.stringify(urn + "-" + size)
```